discovery-provider/src/utils/helpers.py

```diff
@@ -526,6 +531,33 @@ def get_solana_tx_owner(meta, idx) -> str:
     )
 
 
+def get_valid_instruction(
+    tx_message: TransactionMessage, meta: ResultMeta, program_address: str
+) -> Optional[TransactionMessageInstruction]:
+    """Checks that the tx is valid
+    checks for the transaction message for correct instruction log
+    checks accounts keys for claimable token program
+    """
+    account_keys = tx_message["accountKeys"]
+    instructions = tx_message["instructions"]
+    program_index = account_keys.index(program_address)
+    for instruction in instructions:
+        if instruction["programIdIndex"] == program_index:
+            return instruction
+
+    return None
+
+
+def has_instruction(meta: ResultMeta, instruction: str):
+    return any(log == instruction for log in meta["logMessages"])
+
+
+# The transaction might list sender/receiver in a different order in the pubKeys.
+# The "accounts" field of the instruction has the mapping of accounts to pubKey index
```

Comment:
nit: would be rly nice to include a sample of this account structure in the doc here